File: DoohClick.Web/src/app/crm/advertiser/service/advertiser.service.ts
import { Injectable } from "@angular/core";
import { WebApiService } from "../../../shared/service/web-api.service";
import { MvGridParamOption } from "../../../shared/model/param.model";
import { MvAdvertiser, MvAdvertiserDdl, MvAdvertiserFilterOptions, MvTenantIdParam } from "../model/advertiser.model";
import { Observable } from "rxjs";
import { MvGridResponse, MvResponse } from "../../../shared/model/response.model";

@Injectable({
    providedIn: 'root'
})

export class AdvertiserService {

    constructor(
        private api: WebApiService
    )
    {

    }

    getGrid(param: MvGridParamOption<MvAdvertiserFilterOptions>): Observable<MvResponse<MvGridResponse<MvAdvertiser>>> {
        return this.api.get('crm/advertiser/grid', param, true);
    }

    getDdl(param: MvTenantIdParam): Observable<MvResponse<MvAdvertiserDdl[]>> {
        return this.api.get('crm/advertiser/ddl', param);
    }
}

File: DoohClick.API/DoohClick.Service/Application/crm/advertiser/AdvertiserService.cs


using DoohClick.DataAccess.Dapper;
using DoohClick.Interface.Application.crm.advertiser;
using DoohClick.Interface.Shared.JsonSerializer;
using DoohClick.Model.Application.crm.advertiser;
using DoohClick.Model.Shared.Param;
using DoohClick.Model.Shared.Response;
using DoohClick.Service.Shared.Base;
using Newtonsoft.Json;

namespace DoohClick.Service.Application.crm.advertiser
{
    public class AdvertiserService : BaseService, IAdvertiserService
    {
        public AdvertiserService(
                IDataAccessService dataAccessService,
                IJsonSerializer jsonSerializer
            ): base( dataAccessService, jsonSerializer ) 
        { }

        public async Task<MvGridResponse<MvAdvertiser>?> GetAll(MvGridParamOption<MvAdvertiserFilterOptions> param)
        {
            string result = await _dataAccessService.RetrievalProcedure("crm.sp_advertiser_sel", JsonConvert.SerializeObject(param));
            return _jsonSerializer.DeserializeObject<MvGridResponse<MvAdvertiser>>(result);
        }

        public async Task<List<MvAdvertiserDdl>?> GetDdl(MvTenantIdParam param)
        {
            string result = await _dataAccessService.RetrievalProcedure("crm.sp_advertiser_ddl", JsonConvert.SerializeObject(param));
            return _jsonSerializer.DeserializeObject<List<MvAdvertiserDdl>>(result);
        }
    }
}
